src/components/PrivateChat/PrivateChat.tsx

```diff
@@ -280,6 +280,8 @@ class PrivateChat {
         }
     }
     updateModeratorBanner() {
+        if (!this.banner) { return; }
+
         if (this.player.ui_class.match(/moderator/)) {  // surely would be better to use player.is_moderator, but not available!
             this.banner.removeClass("banner-inactive");
```

Comment:
This would throw an error if you removed that condition and it was true, no?

Reply:
Yes that's why under some circumstances at the moment the chat doesn't actually pop up when it should - somehow a chat is being received before this.banner is set.

Reply:
Hah yeah, sorry about that, I obviously should refrain from reviewing first thing in the morning. Thanks for the fix!